### DISCUSSION TITLE
Creating one snippet file to target multiple languages

### DISCUSSION THREAD

**Discussion by alimehasin** [zed-industries/zed#15809] [Feature Requests]
Some snippets applies to more than one languages, specifically in the `.ts` `.js` `.mjs`, `tsx`, etc.., reading the docs, this doesn't seem to be possible at least for me

https://zed.dev/docs/snippets

**Comment by bajrangCoder** [comment]
There is no way to apply the same snippet file for different languages. Copy the same snippet file and rename it for languages you want like:

```
javascript.json
typescript.json
and so on...
```

**Comment by failable** [comment]
The snippet package of Emacs `yasnippet`  has some inheritance feature https://joaotavora.github.io/yasnippet/snippet-organization.html#org7468fa9 . I wish we could have similar function in Zed too.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
